# Perguntas

> 1) Qual pais teve mais casos?
>>1.1) Qual pais teve mais casos fatais?
>>>1.1.1) Conforme a resposta anterior, qual é a atividade que teve mais casos fatais?

>2) Em que ano teve mais casos?
>>2.1) Qual pais teve mais casos fatais no ano referente ao resultado 2)

>> 2.2) Em que ano teve mais casos fatais?

>> 2.3) Qual pais teve mais casos fatais no ano referente ao resultado do 2.2)

>3) Qual é a relação entre o sexo do tubarão e a fatalidade?

# Limpando e tratando os dados

In [1]:
import warnings
warnings.simplefilter("ignore")

In [2]:
import pandas as pd
import numpy as np
import re

In [3]:
df=pd.read_csv('attacks.csv',sep=',')

In [4]:
df

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
25718         NaN          NaN     NaN         NaN        NaN   
25719         NaN          NaN     NaN         NaN        NaN   
25720         NaN          NaN     NaN         NaN        NaN   
25721         NaN          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
25718              NaN                             NaN          NaN   
25719              NaN                             NaN          NaN   
25720              NaN                             NaN          NaN   
25721              NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson�McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
25718              NaN  NaN  ...              NaN   
25719              NaN  NaN  ...              NaN   
25720              NaN  NaN  ...              NaN   
25721              NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
25718                             NaN                       NaN   
25719                             NaN                       NaN   
25720                             NaN                       NaN   
25721                             NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
25718                                                NaN   
25719                                                NaN   
25720                                                NaN   
25721                                                NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [5]:
#Tratamento de valores nulos
df.loc[(df.isna().sum(axis=1))>=24]

Case Number Date  Year Type Country Area Location Activity Name Sex   \
8702          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8703          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8704          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8705          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
8706          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
...           ...  ...   ...  ...     ...  ...      ...      ...  ...  ...   
25717         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25718         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25719         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25720         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
25721         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   

       ... Species  Investigator or Source  pdf href formula href  \
8702   ...      NaN                    NaN  NaN          NaN  NaN   
8703   ...      NaN                    NaN  NaN          NaN  NaN   
8704   ...      NaN                    NaN  NaN          NaN  NaN   
8705   ...      NaN                    NaN  NaN          NaN  NaN   
8706   ...      NaN                    NaN  NaN          NaN  NaN   
...    ...      ...                    ...  ...          ...  ...   
25717  ...      NaN                    NaN  NaN          NaN  NaN   
25718  ...      NaN                    NaN  NaN          NaN  NaN   
25719  ...      NaN                    NaN  NaN          NaN  NaN   
25720  ...      NaN                    NaN  NaN          NaN  NaN   
25721  ...      NaN                    NaN  NaN          NaN  NaN   

      Case Number.1 Case Number.2 original order Unnamed: 22 Unnamed: 23  
8702            NaN           NaN            NaN         NaN         NaN  
8703            NaN           NaN            NaN         NaN         NaN  
8704            NaN           NaN            NaN         NaN         NaN  
8705            NaN           NaN            NaN         NaN         NaN  
8706            NaN           NaN            NaN         NaN         NaN  
...             ...           ...            ...         ...         ...  
25717           NaN           NaN            NaN         NaN         NaN  
25718           NaN           NaN            NaN         NaN         NaN  
25719           NaN           NaN            NaN         NaN         NaN  
25720           NaN           NaN            NaN         NaN         NaN  
25721           NaN           NaN            NaN         NaN         NaN  

[17020 rows x 24 columns]

In [6]:
df=df.dropna(how='all')
df

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
8698            0          NaN     NaN         NaN        NaN   
8699            0          NaN     NaN         NaN        NaN   
8700            0          NaN     NaN         NaN        NaN   
8701            0          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
8698               NaN                             NaN          NaN   
8699               NaN                             NaN          NaN   
8700               NaN                             NaN          NaN   
8701               NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson�McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
8698               NaN  NaN  ...              NaN   
8699               NaN  NaN  ...              NaN   
8700               NaN  NaN  ...              NaN   
8701               NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
8698                              NaN                       NaN   
8699                              NaN                       NaN   
8700                              NaN                       NaN   
8701                              NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
8698                                                 NaN   
8699                                                 NaN   
8700                                                 NaN   
8701                                                 NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [7]:
df=df.rename({'Sex ':'Sex'},axis=1)
df=df.rename({'Species ':'Species'},axis=1)

In [8]:
#Tratando 'Case Number' que tem valor '0' e 'xx'
df=df.loc[(df['Case Number'] != '0') & (df['Case Number'] != 'xx'),: ]
df

Case Number         Date    Year        Type             Country  \
0     2018.06.25  25-Jun-2018  2018.0     Boating                 USA   
1     2018.06.18  18-Jun-2018  2018.0  Unprovoked                 USA   
2     2018.06.09  09-Jun-2018  2018.0     Invalid                 USA   
3     2018.06.08  08-Jun-2018  2018.0  Unprovoked           AUSTRALIA   
4     2018.06.04  04-Jun-2018  2018.0    Provoked              MEXICO   
...          ...          ...     ...         ...                 ...   
6297     ND.0005  Before 1903     0.0  Unprovoked           AUSTRALIA   
6298     ND.0004  Before 1903     0.0  Unprovoked           AUSTRALIA   
6299     ND.0003    1900-1905     0.0  Unprovoked                 USA   
6300     ND.0002    1883-1889     0.0  Unprovoked              PANAMA   
6301     ND.0001    1845-1853     0.0  Unprovoked  CEYLON (SRI LANKA)   

                   Area                             Location      Activity  \
0            California          Oceanside, San Diego County      Paddling   
1               Georgia       St. Simon Island, Glynn County      Standing   
2                Hawaii                         Habush, Oahu       Surfing   
3       New South Wales                   Arrawarra Headland       Surfing   
4                Colima                             La Ticla   Free diving   
...                 ...                                  ...           ...   
6297  Western Australia                          Roebuck Bay        Diving   
6298  Western Australia                                  NaN  Pearl diving   
6299     North Carolina                       Ocracoke Inlet      Swimming   
6300                NaN                 Panama Bay 8�N, 79�W           NaN   
6301   Eastern Province  Below the English fort, Trincomalee      Swimming   

                       Name Sex  ...          Species  \
0               Julie Wolfe   F  ...      White shark   
1           Adyson�McNeely    F  ...              NaN   
2               John Denges   M  ...              NaN   
3                      male   M  ...        2 m shark   
4            Gustavo Ramos    M  ...  Tiger shark, 3m   
...                     ...  ..  ...              ...   
6297                   male   M  ...              NaN   
6298                  Ahmun   M  ...              NaN   
6299  Coast Guard personnel   M  ...              NaN   
6300        Jules Patterson   M  ...              NaN   
6301                   male   M  ...              NaN   

                     Investigator or Source                             pdf  \
0                          R. Collier, GSAF            2018.06.25-Wolfe.pdf   
1            K.McMurray, TrackingSharks.com          2018.06.18-McNeely.pdf   
2            K.McMurray, TrackingSharks.com           2018.06.09-Denges.pdf   
3                            B. Myatt, GSAF        2018.06.08-Arrawarra.pdf   
4                                 A .Kipper            2018.06.04-Ramos.pdf   
...                                     ...                             ...   
6297       H. Taunton; N. Bartlett,  p. 234          ND-0005-RoebuckBay.pdf   
6298  H. Taunton; N. Bartlett,  pp. 233-234               ND-0004-Ahmun.pdf   
6299   F. Schwartz, p.23; C. Creswell, GSAF  ND-0003-Ocracoke_1900-1905.pdf   
6300                    The Sun, 10/20/1938      ND-0002-JulesPatterson.pdf   
6301                             S.W. Baker              ND-0001-Ceylon.pdf   

                                           href formula  \
0     http://sharkattackfile.net/spreadsheets/pdf_di...   
1     http://sharkattackfile.net/spreadsheets/pdf_di...   
2     http://sharkattackfile.net/spreadsheets/pdf_di...   
3     http://sharkattackfile.net/spreadsheets/pdf_di...   
4     http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                 ...   
6297  http://sharkattackfile.net/spreadsheets/pdf_di...   
6298  http://sharkattackfile.net/spreadsheets/pdf_di...   
6299  http://sharkattackfi

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6302 entries, 0 to 6301
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             6301 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6300 non-null   float64
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     5737 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null   

In [10]:
#Conferindo valors nulos no Unnamed 22 e 23 (como tem descrições relevantes precisa ser mantido)
df.loc[df['Unnamed: 22'].isnull()==False]

Case Number         Date    Year        Type Country    Area  \
1478  2006.05.27  27-May-2006  2006.0  Unprovoked     USA  Hawaii   

                Location Activity          Name Sex  ... Species  \
1478  North Shore, O'ahu  Surfing  Bret Desmond   M  ...     NaN   

     Investigator or Source                     pdf  \
1478             R. Collier  2006.05.27-Desmond.pdf   

                                           href formula  \
1478  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
1478  http://sharkattackfile.net/spreadsheets/pdf_di...    2006.05.27   

     Case Number.2 original order   Unnamed: 22 Unnamed: 23  
1478    2006.05.27         4825.0  stopped here         NaN  

[1 rows x 24 columns]

In [11]:
df.loc[df['Unnamed: 23'].isnull()==False]

Case Number                  Date    Year        Type  \
4415    1952.03.30           30-Mar-1952  1952.0  Unprovoked   
5840  1878.09.14.R  Reported 14-Sep-1878  1878.0    Provoked   

                   Country         Area                    Location  \
4415  NETHERLANDS ANTILLES      Curacao                         NaN   
5840                   USA  Connecticut  Branford, New Haven County   

                                             Activity              Name Sex  \
4415  Went to aid of child being menaced by the shark       A.J. Eggink   M   
5840                                          Fishing  Captain Pattison   M   

      ...                                            Species  \
4415  ...  Bull shark, 2.7 m [9'] was captured & dragged ...   
5840  ...                                                NaN   

                                 Investigator or Source  \
4415  J. Randall, p.352 in Sharks & Survival; H.D. B...   
5840                       St. Joseph Herald, 9/14/1878   

                            pdf  \
4415      1952.03.30-Eggink.pdf   
5840  1878.09.14.R-Pattison.pdf   

                                           href formula  \
4415  http://sharkattackfile.net/spreadsheets/pdf_di...   
5840  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
4415  http://sharkattackfile.net/spreadsheets/pdf_di...    1952.03.30   
5840  http://sharkattackfile.net/spreadsheets/pdf_di...  1878.09.14.R   

     Case Number.2 original order Unnamed: 22      Unnamed: 23  
4415    1952.03.30         1888.0         NaN           Teramo  
5840  1878.09.14.R          463.0         NaN  change filename  

[2 rows x 24 columns]

In [12]:
df.drop('Unnamed: 22',axis=1,inplace=True)

In [13]:
df.drop('Unnamed: 23',axis=1,inplace=True)

In [14]:
#Conferindo se tem duplicatas
df.duplicated().value_counts()

False    6302
dtype: int64

In [15]:
#tratamento de coluna 'Sex'
df['Sex'].value_counts()

M      5094
F       637
M         2
N         2
lli       1
.         1
Name: Sex, dtype: int64

In [16]:
df['Sex']=np.where(df['Sex']=='M ','M',df['Sex'])
df['Sex']=np.where(df['Sex']=='N',0,df['Sex'])
df['Sex']=np.where(df['Sex']=='lli',0,df['Sex'])
df['Sex']=np.where(df['Sex']=='.',0,df['Sex'])
df['Sex']=np.where(df['Sex'].isnull()==True,0,df['Sex'])

In [17]:
df['Sex'].value_counts()

M    5096
F     637
0     569
Name: Sex, dtype: int64

In [18]:
df

Case Number         Date    Year        Type             Country  \
0     2018.06.25  25-Jun-2018  2018.0     Boating                 USA   
1     2018.06.18  18-Jun-2018  2018.0  Unprovoked                 USA   
2     2018.06.09  09-Jun-2018  2018.0     Invalid                 USA   
3     2018.06.08  08-Jun-2018  2018.0  Unprovoked           AUSTRALIA   
4     2018.06.04  04-Jun-2018  2018.0    Provoked              MEXICO   
...          ...          ...     ...         ...                 ...   
6297     ND.0005  Before 1903     0.0  Unprovoked           AUSTRALIA   
6298     ND.0004  Before 1903     0.0  Unprovoked           AUSTRALIA   
6299     ND.0003    1900-1905     0.0  Unprovoked                 USA   
6300     ND.0002    1883-1889     0.0  Unprovoked              PANAMA   
6301     ND.0001    1845-1853     0.0  Unprovoked  CEYLON (SRI LANKA)   

                   Area                             Location      Activity  \
0            California          Oceanside, San Diego County      Paddling   
1               Georgia       St. Simon Island, Glynn County      Standing   
2                Hawaii                         Habush, Oahu       Surfing   
3       New South Wales                   Arrawarra Headland       Surfing   
4                Colima                             La Ticla   Free diving   
...                 ...                                  ...           ...   
6297  Western Australia                          Roebuck Bay        Diving   
6298  Western Australia                                  NaN  Pearl diving   
6299     North Carolina                       Ocracoke Inlet      Swimming   
6300                NaN                 Panama Bay 8�N, 79�W           NaN   
6301   Eastern Province  Below the English fort, Trincomalee      Swimming   

                       Name Sex  ... Fatal (Y/N)           Time  \
0               Julie Wolfe   F  ...           N          18h00   
1           Adyson�McNeely    F  ...           N  14h00  -15h00   
2               John Denges   M  ...           N          07h45   
3                      male   M  ...           N            NaN   
4            Gustavo Ramos    M  ...           N            NaN   
...                     ...  ..  ...         ...            ...   
6297                   male   M  ...           Y            NaN   
6298                  Ahmun   M  ...           Y            NaN   
6299  Coast Guard personnel   M  ...           Y            NaN   
6300        Jules Patterson   M  ...           Y            NaN   
6301                   male   M  ...           Y            NaN   

              Species                 Investigator or Source  \
0         White shark                       R. Collier, GSAF   
1                 NaN         K.McMurray, TrackingSharks.com   
2                 NaN         K.McMurray, TrackingSharks.com   
3           2 m shark                         B. Myatt, GSAF   
4     Tiger shark, 3m                              A .Kipper   
...               ...                                    ...   
6297              NaN       H. Taunton; N. Bartlett,  p. 234   
6298              NaN  H. Taunton; N. Bartlett,  pp. 233-234   
6299              NaN   F. Schwartz, p.23; C. Creswell, GSAF   
6300              NaN                    The Sun, 10/20/1938   
6301              NaN                             S.W. Baker   

                                 pdf  \
0               2018.06.25-Wolfe.pdf   
1             2018.06.18-McNeely.pdf   
2              2018.06.09-Denges.pdf   
3           2018.06.08-Arrawarra.pdf   
4               2018.06.04-Ramos.pdf   
...                              ...   
6297          ND-0005-RoebuckBay.pdf   
6298               ND-0004-Ahmun.pdf   
6299  ND-0003-Ocracoke_1900-1905.pdf   
6300      ND-0002-JulesPatterson.pdf   
6301              ND-0001-Ceylon.pdf   

                                           href formula  \
0     http://sharkattackfile.net/spreadsheets/pdf_di...   
1     http://shark

In [19]:
#Tratamento de coluna 'Country'
country_list=list(df['Country'].unique())
country_list

['USA',
 'AUSTRALIA',
 'MEXICO',
 'BRAZIL',
 'ENGLAND',
 'SOUTH AFRICA',
 'THAILAND',
 'COSTA RICA',
 'MALDIVES',
 'BAHAMAS',
 'NEW CALEDONIA',
 'ECUADOR',
 'MALAYSIA',
 'LIBYA',
 nan,
 'CUBA',
 'MAURITIUS',
 'NEW ZEALAND',
 'SPAIN',
 'SAMOA',
 'SOLOMON ISLANDS',
 'JAPAN',
 'EGYPT',
 'ST HELENA, British overseas territory',
 'COMOROS',
 'REUNION',
 'FRENCH POLYNESIA',
 'UNITED KINGDOM',
 'UNITED ARAB EMIRATES',
 'PHILIPPINES',
 'INDONESIA',
 'CHINA',
 'COLUMBIA',
 'CAPE VERDE',
 'Fiji',
 'DOMINICAN REPUBLIC',
 'CAYMAN ISLANDS',
 'ARUBA',
 'MOZAMBIQUE',
 'FIJI',
 'PUERTO RICO',
 'ITALY',
 'ATLANTIC OCEAN',
 'GREECE',
 'ST. MARTIN',
 'FRANCE',
 'PAPUA NEW GUINEA',
 'TRINIDAD & TOBAGO',
 'KIRIBATI',
 'ISRAEL',
 'DIEGO GARCIA',
 'TAIWAN',
 'JAMAICA',
 'PALESTINIAN TERRITORIES',
 'GUAM',
 'SEYCHELLES',
 'BELIZE',
 'NIGERIA',
 'TONGA',
 'SCOTLAND',
 'CANADA',
 'CROATIA',
 'SAUDI ARABIA',
 'CHILE',
 'ANTIGUA',
 'KENYA',
 'RUSSIA',
 'TURKS & CAICOS',
 'UNITED ARAB EMIRATES (UAE)',
 'AZORES',
 

In [20]:
country_list=['USA',
 'AUSTRALIA',
 'MEXICO',
 'BRAZIL',
 'ENGLAND',
 'SOUTH AFRICA',
 'THAILAND',
 'COSTA RICA',
 'MALDIVES',
 'BAHAMAS',
 'NEW CALEDONIA',
 'ECUADOR',
 'MALAYSIA',
 'LIBYA',
 'CUBA',
 'MAURITIUS',
 'NEW ZEALAND',
 'SPAIN',
 'SAMOA',
 'SOLOMON ISLANDS',
 'JAPAN',
 'EGYPT',
 'ST HELENA, British overseas territory',
 'COMOROS',
 'REUNION',
 'FRENCH POLYNESIA',
 'UNITED KINGDOM',
 'UNITED ARAB EMIRATES',
 'PHILIPPINES',
 'INDONESIA',
 'CHINA',
 'COLUMBIA',
 'CAPE VERDE',
 'Fiji',
 'DOMINICAN REPUBLIC',
 'CAYMAN ISLANDS',
 'ARUBA',
 'MOZAMBIQUE',
 'FIJI',
 'PUERTO RICO',
 'ITALY',
 'ATLANTIC OCEAN',
 'GREECE',
 'ST. MARTIN',
 'FRANCE',
 'PAPUA NEW GUINEA',
 'TRINIDAD & TOBAGO',
 'KIRIBATI',
 'ISRAEL',
 'DIEGO GARCIA',
 'TAIWAN',
 'JAMAICA',
 'PALESTINIAN TERRITORIES',
 'GUAM',
 'SEYCHELLES',
 'BELIZE',
 'NIGERIA',
 'TONGA',
 'SCOTLAND',
 'CANADA',
 'CROATIA',
 'SAUDI ARABIA',
 'CHILE',
 'ANTIGUA',
 'KENYA',
 'RUSSIA',
 'TURKS & CAICOS',
 'UNITED ARAB EMIRATES (UAE)',
 'AZORES',
 'SOUTH KOREA',
 'MALTA',
 'VIETNAM',
 'MADAGASCAR',
 'PANAMA',
 'SOMALIA',
 'NEVIS',
 'BRITISH VIRGIN ISLANDS',
 'NORWAY',
 'SENEGAL',
 'YEMEN',
 'GULF OF ADEN',
 'Sierra Leone',
 'ST. MAARTIN',
 'GRAND CAYMAN',
 'Seychelles',
 'LIBERIA',
 'VANUATU',
 'MEXICO ',
 'HONDURAS',
 'VENEZUELA',
 'SRI LANKA',
 ' TONGA',
 'URUGUAY',
 'INDIA',
 'MICRONESIA',
 'CARIBBEAN SEA',
 'OKINAWA',
 'TANZANIA',
 'MARSHALL ISLANDS',
 'EGYPT / ISRAEL',
 'NORTHERN ARABIAN SEA',
 'HONG KONG',
 'EL SALVADOR',
 'ANGOLA',
 'BERMUDA',
 'MONTENEGRO',
 'IRAN',
 'TUNISIA',
 'NAMIBIA',
 'NORTH ATLANTIC OCEAN',
 'PORTUGAL',
 'SOUTH CHINA SEA',
 'BANGLADESH',
 'PALAU',
 'WESTERN SAMOA',
 'PACIFIC OCEAN ',
 'BRITISH ISLES',
 'GRENADA',
 'IRAQ',
 'TURKEY',
 'SINGAPORE',
 'NEW BRITAIN',
 'SUDAN',
 'JOHNSTON ISLAND',
 'SOUTH PACIFIC OCEAN',
 'NEW GUINEA',
 'RED SEA',
 'NORTH PACIFIC OCEAN',
 'FEDERATED STATES OF MICRONESIA',
 'MID ATLANTIC OCEAN',
 'ADMIRALTY ISLANDS',
 'BRITISH WEST INDIES',
 'SOUTH ATLANTIC OCEAN',
 'PERSIAN GULF',
 'RED SEA / INDIAN OCEAN',
 'PACIFIC OCEAN',
 'NORTH SEA',
 'NICARAGUA ',
 'MALDIVE ISLANDS',
 'AMERICAN SAMOA',
 'ANDAMAN / NICOBAR ISLANDAS',
 'GABON',
 'MAYOTTE',
 'NORTH ATLANTIC OCEAN ',
 'THE BALKANS',
 'SUDAN?',
 'ARGENTINA',
 'MARTINIQUE',
 'INDIAN OCEAN',
 'GUATEMALA',
 'NETHERLANDS ANTILLES',
 'NORTHERN MARIANA ISLANDS',
 'IRAN / IRAQ',
 'JAVA',
 'SIERRA LEONE',
 ' PHILIPPINES',
 'NICARAGUA',
 'CENTRAL PACIFIC',
 'SOLOMON ISLANDS / VANUATU',
 'SOUTHWEST PACIFIC OCEAN',
 'BAY OF BENGAL',
 'MID-PACIFC OCEAN',
 'SLOVENIA',
 'CURACAO',
 'ICELAND',
 'ITALY / CROATIA',
 'BARBADOS',
 'MONACO',
 'GUYANA',
 'HAITI',
 'SAN DOMINGO',
 'IRELAND',
 'KUWAIT',
 'YEMEN ',
 'REUNION ISLAND',
 'FALKLAND ISLANDS',
 'CRETE',
 'CYPRUS',
 'EGYPT ',
 'WEST INDIES',
 'BURMA',
 'LEBANON',
 'PARAGUAY',
 'BRITISH NEW GUINEA',
 'CEYLON',
 'OCEAN',
 'GEORGIA',
 'SYRIA',
 'TUVALU',
 'INDIAN OCEAN?',
 'GUINEA',
 'ANDAMAN ISLANDS',
 'EQUATORIAL GUINEA / CAMEROON',
 'COOK ISLANDS',
 'TOBAGO',
 'PERU',
 'AFRICA',
 'ALGERIA',
 'Coast of AFRICA',
 'TASMAN SEA',
 'GHANA',
 'GREENLAND',
 'MEDITERRANEAN SEA',
 'SWEDEN',
 'ROATAN',
 'Between PORTUGAL & INDIA',
 'DJIBOUTI',
 'BAHREIN',
 'KOREA',
 'RED SEA?',
 'ASIA?',
 'CEYLON (SRI LANKA)']

In [21]:
country_list.sort()
country_list

[' PHILIPPINES',
 ' TONGA',
 'ADMIRALTY ISLANDS',
 'AFRICA',
 'ALGERIA',
 'AMERICAN SAMOA',
 'ANDAMAN / NICOBAR ISLANDAS',
 'ANDAMAN ISLANDS',
 'ANGOLA',
 'ANTIGUA',
 'ARGENTINA',
 'ARUBA',
 'ASIA?',
 'ATLANTIC OCEAN',
 'AUSTRALIA',
 'AZORES',
 'BAHAMAS',
 'BAHREIN',
 'BANGLADESH',
 'BARBADOS',
 'BAY OF BENGAL',
 'BELIZE',
 'BERMUDA',
 'BRAZIL',
 'BRITISH ISLES',
 'BRITISH NEW GUINEA',
 'BRITISH VIRGIN ISLANDS',
 'BRITISH WEST INDIES',
 'BURMA',
 'Between PORTUGAL & INDIA',
 'CANADA',
 'CAPE VERDE',
 'CARIBBEAN SEA',
 'CAYMAN ISLANDS',
 'CENTRAL PACIFIC',
 'CEYLON',
 'CEYLON (SRI LANKA)',
 'CHILE',
 'CHINA',
 'COLUMBIA',
 'COMOROS',
 'COOK ISLANDS',
 'COSTA RICA',
 'CRETE',
 'CROATIA',
 'CUBA',
 'CURACAO',
 'CYPRUS',
 'Coast of AFRICA',
 'DIEGO GARCIA',
 'DJIBOUTI',
 'DOMINICAN REPUBLIC',
 'ECUADOR',
 'EGYPT',
 'EGYPT ',
 'EGYPT / ISRAEL',
 'EL SALVADOR',
 'ENGLAND',
 'EQUATORIAL GUINEA / CAMEROON',
 'FALKLAND ISLANDS',
 'FEDERATED STATES OF MICRONESIA',
 'FIJI',
 'FRANCE',
 'FRENCH PO

In [22]:
df['Country']=np.where(df['Country']==' PHILIPPINES','PHILIPPINES',df['Country'])
df['Country']=np.where(df['Country']==' TONGA','TONGA',df['Country'])
df['Country']=np.where(df['Country']=='ASIA?','ASIA',df['Country'])
df['Country']=np.where(df['Country']=='CEYLON (SRI LANKA)','CEYLON',df['Country'])
df['Country']=np.where(df['Country']=='EGYPT ','EGYPT',df['Country'])
df['Country']=np.where(df['Country']=='MALDIVE ISLANDS','MALDIVES',df['Country'])
df['Country']=np.where(df['Country']=='MEXICO ','MEXICO',df['Country'])
df['Country']=np.where(df['Country']=='NICARAGUA ','NICARAGUA',df['Country'])
df['Country']=np.where(df['Country']=='NORTH ATLANTIC OCEAN ','NORTH ATLANTIC OCEAN',df['Country'])
df['Country']=np.where(df['Country']=='PACIFIC OCEAN ','PACIFIC OCEAN',df['Country'])
df['Country']=np.where(df['Country']=='RED SEA?','RED SEA',df['Country'])
df['Country']=np.where(df['Country']=='REUNION ISLAND','REUNION',df['Country'])
df['Country']=np.where(df['Country']=='KOREA','SOUTH KOREA',df['Country'])
df['Country']=np.where(df['Country']=='ST. MAARTIN','ST. MARTIN',df['Country'])
df['Country']=np.where(df['Country']=='SUDAN?','SUDAN',df['Country'])
df['Country']=np.where(df['Country']=='UNITED ARAB EMIRATES (UAE)','UNITED ARAB EMIRATES',df['Country'])
df['Country']=np.where(df['Country']=='YEMEN ','YEMEN',df['Country'])

In [23]:
df['Year'].isnull().value_counts()

False    6300
True        2
Name: Year, dtype: int64

In [24]:
df.loc[df['Year'].isnull()==True,:]

Case Number                  Date  Year        Type    Country  \
187   2017.01.08.R  Reported 08-Jan-2017   NaN     Invalid  AUSTRALIA   
6079  1836.08.19.R  Reported 19-Aug-1836   NaN  Unprovoked    ENGLAND   

            Area    Location      Activity          Name Sex  ... Fatal (Y/N)  \
187   Queensland         NaN  Spearfishing  Kerry Daniel   M  ...         NaN   
6079  Cumberland  Whitehaven      Swimming         a boy   M  ...           Y   

     Time     Species  Investigator or Source                           pdf  \
187   NaN  Bull shark  Liquid Vision 1/8/2017  2017.01.08.R-KerryDaniel.pdf   
6079  NaN         NaN          C. Moore, GSAF   1835.08.19.R-Whitehaven.pdf   

                                           href formula  \
187   http://sharkattackfile.net/spreadsheets/pdf_di...   
6079  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
187   http://sharkattackfile.net/spreadsheets/pdf_di...  2017.01.08.R   
6079  http://sharkattackfile.net/spreadsheets/pdf_di...  1836.08.19.R   

     Case Number.2 original order  
187   2017.01.08.R         6116.0  
6079  1836.08.19.R          224.0  

[2 rows x 22 columns]

In [25]:
#transformando ano conforme ao Date
df['Year']=np.where((df['Year'].isnull()==True) & (df['Type']=='Invalid'),2017,df['Year']) 
df['Year']=np.where((df['Year'].isnull()==True) & (df['Type']=='Unprovoked'),1836,df['Year'])


In [26]:
df.Year=df.Year.astype(int)

In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6302 entries, 0 to 6301
Data columns (total 22 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             6301 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6302 non-null   int32  
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     6302 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null   

In [28]:
df['Fatal (Y/N)'].value_counts()

N          4293
Y          1388
UNKNOWN      71
 N            7
y             1
M             1
N             1
2017          1
Name: Fatal (Y/N), dtype: int64

In [29]:
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']==' N','N',df['Fatal (Y/N)'])
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='y','Y',df['Fatal (Y/N)'])
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='N ','N',df['Fatal (Y/N)'])
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='M','N',df['Fatal (Y/N)']) #Erro de digitação
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='2017','N',df['Fatal (Y/N)']) #Pelo contexto não parece ser fatal
df['Fatal (Y/N)']=np.where(df['Fatal (Y/N)']=='UNKNOWN',0,df['Fatal (Y/N)']) #transformar unknown para o

In [30]:
df_filtered=df[['Year','Country','Activity','Sex','Injury','Fatal (Y/N)','Species']]

In [31]:
df_filtered

Year    Country      Activity Sex  \
0     2018        USA      Paddling   F   
1     2018        USA      Standing   F   
2     2018        USA       Surfing   M   
3     2018  AUSTRALIA       Surfing   M   
4     2018     MEXICO   Free diving   M   
...    ...        ...           ...  ..   
6297     0  AUSTRALIA        Diving   M   
6298     0  AUSTRALIA  Pearl diving   M   
6299     0        USA      Swimming   M   
6300     0     PANAMA           NaN   M   
6301     0     CEYLON      Swimming   M   

                                                 Injury Fatal (Y/N)  \
0     No injury to occupant, outrigger canoe and pad...           N   
1                            Minor injury to left thigh           N   
2          Injury to left lower leg from surfboard skeg           N   
3                             Minor injury to lower leg           N   
4     Lacerations to leg & hand shark PROVOKED INCIDENT           N   
...                                                 ...         ...   
6297                                              FATAL           Y   
6298                                              FATAL           Y   
6299                                              FATAL           Y   
6300                                              FATAL           Y   
6301  FATAL. "Shark bit him in half, carrying away t...           Y   

              Species  
0         White shark  
1                 NaN  
2                 NaN  
3           2 m shark  
4     Tiger shark, 3m  
...               ...  
6297              NaN  
6298              NaN  
6299              NaN  
6300              NaN  
6301              NaN  

[6302 rows x 7 columns]

In [32]:
df_filtered.duplicated().value_counts()

False    6251
True       51
dtype: int64

In [33]:
df_filtered=df_filtered.drop_duplicates()
df_filtered

Year    Country      Activity Sex  \
0     2018        USA      Paddling   F   
1     2018        USA      Standing   F   
2     2018        USA       Surfing   M   
3     2018  AUSTRALIA       Surfing   M   
4     2018     MEXICO   Free diving   M   
...    ...        ...           ...  ..   
6297     0  AUSTRALIA        Diving   M   
6298     0  AUSTRALIA  Pearl diving   M   
6299     0        USA      Swimming   M   
6300     0     PANAMA           NaN   M   
6301     0     CEYLON      Swimming   M   

                                                 Injury Fatal (Y/N)  \
0     No injury to occupant, outrigger canoe and pad...           N   
1                            Minor injury to left thigh           N   
2          Injury to left lower leg from surfboard skeg           N   
3                             Minor injury to lower leg           N   
4     Lacerations to leg & hand shark PROVOKED INCIDENT           N   
...                                                 ...         ...   
6297                                              FATAL           Y   
6298                                              FATAL           Y   
6299                                              FATAL           Y   
6300                                              FATAL           Y   
6301  FATAL. "Shark bit him in half, carrying away t...           Y   

              Species  
0         White shark  
1                 NaN  
2                 NaN  
3           2 m shark  
4     Tiger shark, 3m  
...               ...  
6297              NaN  
6298              NaN  
6299              NaN  
6300              NaN  
6301              NaN  

[6251 rows x 7 columns]

# Resultado

 > **1) Qual pais teve mais casos?**

In [34]:
df_filtered

Year    Country      Activity Sex  \
0     2018        USA      Paddling   F   
1     2018        USA      Standing   F   
2     2018        USA       Surfing   M   
3     2018  AUSTRALIA       Surfing   M   
4     2018     MEXICO   Free diving   M   
...    ...        ...           ...  ..   
6297     0  AUSTRALIA        Diving   M   
6298     0  AUSTRALIA  Pearl diving   M   
6299     0        USA      Swimming   M   
6300     0     PANAMA           NaN   M   
6301     0     CEYLON      Swimming   M   

                                                 Injury Fatal (Y/N)  \
0     No injury to occupant, outrigger canoe and pad...           N   
1                            Minor injury to left thigh           N   
2          Injury to left lower leg from surfboard skeg           N   
3                             Minor injury to lower leg           N   
4     Lacerations to leg & hand shark PROVOKED INCIDENT           N   
...                                                 ...         ...   
6297                                              FATAL           Y   
6298                                              FATAL           Y   
6299                                              FATAL           Y   
6300                                              FATAL           Y   
6301  FATAL. "Shark bit him in half, carrying away t...           Y   

              Species  
0         White shark  
1                 NaN  
2                 NaN  
3           2 m shark  
4     Tiger shark, 3m  
...               ...  
6297              NaN  
6298              NaN  
6299              NaN  
6300              NaN  
6301              NaN  

[6251 rows x 7 columns]

In [35]:
df_filtered['Country'].value_counts()

USA                       2208
AUSTRALIA                 1328
SOUTH AFRICA               579
PAPUA NEW GUINEA           134
NEW ZEALAND                128
                          ... 
RED SEA / INDIAN OCEAN       1
IRELAND                      1
SYRIA                        1
SWEDEN                       1
GRAND CAYMAN                 1
Name: Country, Length: 196, dtype: int64

> - *`Resposta : USA teve mais casos (2208 casos)`*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

 > **1.1) Qual pais teve mais casos fatais?**

In [36]:
df_fatal=df_filtered.loc[df_filtered['Fatal (Y/N)']=='Y',:]
df_fatal[['Year','Fatal (Y/N)','Country']].groupby(by=['Fatal (Y/N)','Country']).count().sort_values(by='Year', ascending=False)

Year
Fatal (Y/N) Country               
Y           AUSTRALIA          276
            USA                183
            SOUTH AFRICA       106
            PAPUA NEW GUINEA    56
            MEXICO              43
...                            ...
            INDIAN OCEAN?        1
            CRETE                1
            SAMOA                1
            SAN DOMINGO          1
            AFRICA               1

[142 rows x 1 columns]

In [37]:
total_fatal=len(df_fatal.loc[df_fatal['Fatal (Y/N)']=='Y','Fatal (Y/N)'])
total_fatal

1366

In [38]:
australia_fatal=len(df_fatal.loc[(df_fatal['Fatal (Y/N)']=='Y') & (df_fatal['Country']=='AUSTRALIA'),'Fatal (Y/N)'])
australia_fatal

276

In [39]:
australia_porc=australia_fatal/total_fatal*100
australia_porc

20.20497803806735

> - *`Resposta : Australia teve mais casos fatais (276 casos(20.2% dos casos)`*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

 > **1.1.1) Conforme a resposta anterior qual é a atividade que teve mais casos fatais?**

In [40]:
df_fatal_australia=df_fatal.loc[df_fatal['Country']=='AUSTRALIA',:]
df_fatal_australia=df_fatal_australia.reset_index()
df_fatal_australia

index  Year    Country                  Activity Sex Injury Fatal (Y/N)  \
0      155  2017  AUSTRALIA                   Surfing   F  FATAL           Y   
1      267  2016  AUSTRALIA                    Diving   F  FATAL           Y   
2      271  2016  AUSTRALIA                   Surfing   M  FATAL           Y   
3      385  2015  AUSTRALIA  Scallop diving on hookah   M  FATAL           Y   
4      448  2015  AUSTRALIA                   Surfing   M  FATAL           Y   
..     ...   ...        ...                       ...  ..    ...         ...   
271   6294     0  AUSTRALIA                   Fishing   M  FATAL           Y   
272   6295     0  AUSTRALIA                   Fishing   M  FATAL           Y   
273   6296     0  AUSTRALIA                  Swimming   M  FATAL           Y   
274   6297     0  AUSTRALIA                    Diving   M  FATAL           Y   
275   6298     0  AUSTRALIA              Pearl diving   M  FATAL           Y   

                                               Species  
0                                          White shark  
1                                           3+ m shark  
2                                          White shark  
3                            White shark, 3.9 to 4.2 m  
4                                     3.5 to 4 m shark  
..                                                 ...  
271                                       Blue pointer  
272                                      Blue pointers  
273  Said to involve a grey nurse shark that leapt ...  
274                                                NaN  
275                                                NaN  

[276 rows x 8 columns]

In [41]:
df_activity=df_fatal_australia['Activity'].fillna('0')
df_activity

0                       Surfing
1                        Diving
2                       Surfing
3      Scallop diving on hookah
4                       Surfing
                 ...           
271                     Fishing
272                     Fishing
273                    Swimming
274                      Diving
275                Pearl diving
Name: Activity, Length: 276, dtype: object

In [42]:
t=[]
for i in list(df_activity):
    if re.search('[Bb][Aa][Tt][Hh][Ii][Nn][Gg]',i):
        t.append('Bathing')
    elif re.search('[Ss][Ww][Ii][Mm][Mm][Ii][Nn][Gg]',i):
        t.append('Swimming')
    elif re.search('[Dd][Ii][Vv][Ii][Nn][Gg]',i):
        t.append('Diving')
    elif re.search('[Ff][Ii][Ss][Hh][Ii][Nn][Gg]',i):
        t.append('Fishing')
    elif re.search('[Ss][Uu][Rr][Ff][Ii][Nn][Gg]',i):
        t.append('Surfing')
    elif re.search('[Ff][Ee][Ll][Ll]',i):
        t.append('Fell')
    elif re.search('[Ss][Tt][Aa][Nn][Dd][Ii][Nn][Gg]',i):
        t.append('Standing')
    elif re.search('[Ff][Ll][Oo][Aa][Tt][Ii][Nn][Gg]',i):
        t.append('Floating')
    elif re.search('[Ss][Hh][Ii][Pp][Ww][Rr][Ee][Cc][Kk]',i):
        t.append('Shipwreck')
    elif re.search('[Tt][Rr][Ee][Aa][Dd][Ii][Nn][Gg]',i):
        t.append('Treading')
    elif re.search('[Ww][Aa][Dd][Ii][Nn][Gg]',i):
        t.append('Wading')
    else:
        t.append(i)
t

['Surfing',
 'Diving',
 'Surfing',
 'Diving',
 'Surfing',
 'Fishing',
 'Fishing',
 'Swimming',
 'Swimming',
 'Diving',
 'Body boarding',
 'Surfing',
 'Surfing',
 'Diving',
 'Diving',
 'Swimming',
 'Body boarding',
 'Diving',
 'Surfing',
 'Snorkeling',
 'Body boarding',
 'Swimming',
 'Diving',
 'Snorkeling',
 'Scurfing (surfboard being  towed behind a boat)',
 'Fishing',
 'Surfing',
 'Swimming',
 'Swimming',
 'Diving',
 'Swimming',
 'Surfing',
 'Surfing',
 'Surfing',
 'Diving',
 'Surfing',
 'Diving',
 'Diving',
 'Diving',
 'Diving',
 'Diving',
 'Surfing',
 'Diving',
 'Diving',
 'Surfing',
 'The Christie V sank on 11/6/1988, survivors were adrift on a dinghy',
 'The Christie V sank on 11/6/1988, survivors were adrift on a dinghy',
 'Diving',
 'Diving',
 'Fell',
 'Swimming',
 'Swimming',
 'Surfing',
 'Fishing',
 'Floating',
 'Their 9 m launch was run down by a 25,000-ton Japanese freighter  on the night of 3-11-1977 & they drifted, clinging to an icebox for 2 days',
 'Diving',
 'Swimming'

In [43]:
activity=pd.DataFrame(t).rename({0:'Activity_tratado'},axis=1)
activity

Activity_tratado
0            Surfing
1             Diving
2            Surfing
3             Diving
4            Surfing
..               ...
271          Fishing
272          Fishing
273         Swimming
274           Diving
275           Diving

[276 rows x 1 columns]

In [44]:
df_fatal_australia_activity=pd.concat([df_fatal_australia,activity],axis=1).drop('index',axis=1)
df_fatal_australia_activity

Year    Country                  Activity Sex Injury Fatal (Y/N)  \
0    2017  AUSTRALIA                   Surfing   F  FATAL           Y   
1    2016  AUSTRALIA                    Diving   F  FATAL           Y   
2    2016  AUSTRALIA                   Surfing   M  FATAL           Y   
3    2015  AUSTRALIA  Scallop diving on hookah   M  FATAL           Y   
4    2015  AUSTRALIA                   Surfing   M  FATAL           Y   
..    ...        ...                       ...  ..    ...         ...   
271     0  AUSTRALIA                   Fishing   M  FATAL           Y   
272     0  AUSTRALIA                   Fishing   M  FATAL           Y   
273     0  AUSTRALIA                  Swimming   M  FATAL           Y   
274     0  AUSTRALIA                    Diving   M  FATAL           Y   
275     0  AUSTRALIA              Pearl diving   M  FATAL           Y   

                                               Species Activity_tratado  
0                                          White shark          Surfing  
1                                           3+ m shark           Diving  
2                                          White shark          Surfing  
3                            White shark, 3.9 to 4.2 m           Diving  
4                                     3.5 to 4 m shark          Surfing  
..                                                 ...              ...  
271                                       Blue pointer          Fishing  
272                                      Blue pointers          Fishing  
273  Said to involve a grey nurse shark that leapt ...         Swimming  
274                                                NaN           Diving  
275                                                NaN           Diving  

[276 rows x 8 columns]

In [45]:
df_fatal_australia_activity['Activity_tratado'].value_counts()

Swimming                                                                                                                                                                                                                                                          81
Diving                                                                                                                                                                                                                                                            45
Bathing                                                                                                                                                                                                                                                           26
Surfing                                                                                                                                                                                                                  

In [46]:
total_case=len(df_fatal_australia_activity)
total_case

276

In [47]:
case_swimming=len(df_fatal_australia_activity.loc[df_fatal_australia_activity['Activity_tratado']=='Swimming',:])
case_swimming

81

In [48]:
swimming_percentage=case_swimming/total_case*100
swimming_percentage

29.347826086956523

> - *`Resposta : Na Australia entre 283 casos fatais, 83 casos(29.3%) aconteceram enquanto nadando no mar`*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

 >  2) **Em que ano teve mais casos?**

In [49]:
df_filtered['Year'].value_counts()

2015    143
2017    136
2016    130
2011    128
2014    125
       ... 
1786      1
1758      1
1742      1
1738      1
1554      1
Name: Year, Length: 249, dtype: int64

> - *`Resposta : Em 2015 teve mais casos (143 casos)`*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

>  2.1) **Qual pais teve mais casos fatais no ano referente ao resultado 2)**

In [50]:
df_filtered.loc[df_filtered['Year']==2015,'Country'].value_counts()

USA                 73
AUSTRALIA           31
SOUTH AFRICA        10
REUNION              4
FRENCH POLYNESIA     4
MOZAMBIQUE           3
BAHAMAS              3
SPAIN                2
BRAZIL               2
ECUADOR              1
ITALY                1
COSTA RICA           1
FIJI                 1
ATLANTIC OCEAN       1
PUERTO RICO          1
THAILAND             1
ARUBA                1
MEXICO               1
EGYPT                1
NEW CALEDONIA        1
Name: Country, dtype: int64

> - *`Resposta : USA (73 casos)`*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

>  2.2) **Em que ano teve mais casos fatais?**

In [51]:
df_fatal['Year'].value_counts()[1:]

1959    24
1956    20
1958    18
1944    17
1942    16
        ..
1786     1
1785     1
1784     1
1783     1
2018     1
Name: Year, Length: 219, dtype: int64

> - *`Resposta : Em 1959 teve mais casos fatais`*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

>  2.3) **Qual pais teve mais casos fatais no ano referente ao resultado do 2.2)**

In [ ]:
df_fatal.loc[df_fatal['Year']==1959,:'Country'].value_counts()

> - *`Resposta : Mexico (3 casos), Philippines (3 casos), USA (3 casos) `*

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

>  3) **Qual é a relação entre o sexo do tubarão e a fatalidade**

In [ ]:
df_corr=df_filtered[['Sex','Fatal (Y/N)']]
df_corr

In [ ]:
df_onehotencoding=df_corr
df_onehotencoding['Male']=np.where(df_filtered['Sex']=='M',1,0)
df_onehotencoding['Female']=np.where(df_filtered['Sex']=='F',1,0)

df_onehotencoding['Fatal']=np.where(df_filtered['Fatal (Y/N)']=='Y',True,False)
df_onehotencoding.drop('Sex',axis=1, inplace=True)

In [ ]:
df_onehotencoding

In [ ]:
df_onehotencoding.drop('Fatal (Y/N)',axis=1, inplace=True)

In [ ]:
df_onehotencoding

In [ ]:
df_onehotencoding.corr()

In [ ]:
df_onehotencoding.corr()[['Fatal']]

> - *`Resposta : A fatalidade quase não tem nenhuma relação com sexo do tubarão `*